integration-tests/features/dependency-resolution.feature

```diff
@@ -0,0 +1,39 @@
+Feature: Dependency resolution
+
+  Scenario: launch a environment with dependencies
+    Given environment "3" does not exist
+    When the user launches environment "3"
+    Then all the stacks in environment "3" are in "CREATE_COMPLETE"
+    and that stack "3/A" was created before "3/B"
+    and that stack "3/B" was created before "3/C"
+
+  Scenario: launch a environment with dependencies that is partially complete
+    Given stack "3/A" exists in "CREATE_COMPLETE" state
+    and stack "3/B" exists in "CREATE_COMPLETE" state
+    and stack "3/C" does not exist
+    When the user launches environment "3"
+    Then all the stacks in environment "3" are in "CREATE_COMPLETE"
+    and that stack "3/A" was created before "3/B"
+    and that stack "3/B" was created before "3/C"
+
+  # Need to fix bug to pass the scenario
```

Comment:
Its a bug we've had for a while, so I'd rather just go ahead and get these new tests in. I'll make a issue with the bug and remove these comments from the code.